### ISSUE TITLE
Implementation of SLF4J Should Be Left To Users Of java-nats-streaming

### ISSUE THREAD

**Issue by sfrooster** [nats-io/java-nats-streaming#33]
When used in a library that others will consume (java-nats-streaming, for example), logging implementation should be deferred to the consumer.

SLF4J agrees here: http://www.slf4j.org/codes.html#StaticLoggerBinder

Instead of requiring specific implementations of logging, java-nats-streaming should bind to "slf4j-api".

**Comment by mcqueary** [comment]
Michael, I'm not sure I follow. `java-nats-streaming` doesn't bind to any specific logging implementation. It does implicitly depend on slf4j-api (through `jnats`), but that's it.

When/where did you see otherwise? Maybe an older version of the examples?

**Comment by sfrooster** [comment]
When we call ConnectionFactory.createConnection we get the following:

`SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.`

**Comment by mcqueary** [comment]
I think perhaps you misinterpreted [what you read regarding `StaticLoggerBinder`](http://www.slf4j.org/codes.html#StaticLoggerBinder). The error you're getting is because you have not bound a logger; slf4j prints the error and defaults to a no-op  implementation in this case. As stated on their page, if you don't care about logging and want to suppress this error, just put `slf4j-nop.jar` on your classpath, or if you do care about logging, place one of the many slf4j bindings on your classpath (e.g. log4j, jdk1.4, logback-classic, etc).

**Comment by mcqueary** [comment]
I have updated the README in both `nats-io/jnats` and `nats-io/java-nats-streaming` to include this info.

**Comment by sfrooster** [comment]
Agreed - I finally managed to sort through this and the issue wasn't yours.

**Comment by mcqueary** [comment]
Cool, glad you got it sorted!
